### ISSUE TITLE
How to seamlessly create communication between winforms and webview when handle more number of async requests?

### ISSUE THREAD

**Issue by VigneshSonaiya** [MicrosoftEdge/WebView2Feedback#1341]
I have created winform application with webview2. In my case I need to get the data asynchronously from winform. We know in webview2, we can get the data using addeventlistener method. but how this works when we post multiple messages at same time?

**Comment by darbid** [comment]
Hi @VigneshSonaiya when messaging between your Winform app process and webview 2 there are a number of ways to do it.   Have a look at [Chromium WebView2 Control and .NET to JavaScript Interop - Part 2](https://weblog.west-wind.com/posts/2021/Jan/26/Chromium-WebView2-Control-and-NET-to-JavaScript-Interop-Part-2) which outlines really well your options with examples.  Basically you can "message" the current webpage's top document being displayed in Webview2 with [WebView2.ExecuteScriptAsync(String) Method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.wpf.webview2.executescriptasync?view=webview2-dotnet-1.0.818.41#Microsoft_Web_WebView2_Wpf_WebView2_ExecuteScriptAsync_System_String_).  To send "messages" back from the webpage you have 2 options [CoreWebView2.AddHostObjectToScript(String, Object) Method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.addhostobjecttoscript?view=webview2-dotnet-1.0.818.41#Microsoft_Web_WebView2_Core_CoreWebView2_AddHostObjectToScript_System_String_System_Object_) or [WebView2.WebMessageReceived Event](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.wpf.webview2.webmessagereceived?view=webview2-dotnet-1.0.818.41).

In the case of WebView2.WebMessageReceived Event you would need to not only return the data or message you want but some kind of id so that your Winform .net code knows which message it is.

**Comment by VigneshSonaiya** [comment]
@darbid, Yes I have followed the same way in my application. but my case is like, I built the application which has different sections or widgets to pull the data from winform(which is my backend rightnow). When all sections are trying to get the data from backend at same time. How to get the proper data at this time?

In Cefsharp browser, they have a option to do the [javascript binding](https://github.com/cefsharp/CefSharp/wiki/General-Usage#async-javascript-binding-jsb). So I can call any method I want.

Is there any way kind of this?

How to call method from C# in my browser?
